webapp/channels/src/components/rhs_header_post/rhs_header_post.spec.tsx

```diff
@@ -58,27 +60,27 @@ describe('rhs_header_post', () => {
             },
             posts: {
                 posts: {
-                    42: {
-                        id: 42,
+                    42: TestHelper.getPostMock({
+                        id: '42',
                         message: 'where is @jessica.hyde?',
-                    },
-                    43: {
-                        id: 43,
+                    }),
+                    43: TestHelper.getPostMock({
+                        id: '43',
                         message: 'not a mention',
-                    },
+                    }),
                 },
             },
             threads: {
                 threads: {
                     42: {
-                        id: 42,
+                        id: '42',
                         reply_count: 0,
-                        is_following: null,
+                        is_following: null as any, // This is supposed to be boolean based on the type definitions, but this relies on this being null in some cases
```

Comment:
Can we just say `boolean | null`? Or is the plan to remove the `null` type altogether? Or does it makes sense to just make this explicitly false for now (since I assume we're just checking for falsy)

Reply:
I'd rather not change that right now since there's something really bizarre going on with the types here which feels like it would need bigger changes to address. From what I can tell, this field is added by the client to a response from the server, and in some cases it's null and others it's unset, so I don't know if that produces a meaningful difference or not. I left this comment on it to hopefully help someone in the future understand that it's weird, but I'd rather not touch this right now